pecos/xmc/xlinear/predict.py

```diff
@@ -116,30 +125,55 @@ def do_predict(args):
     # Load data
     Xt = XLinearModel.load_feature_matrix(args.inst_path)
 
-    # Model Predicting
-    xlinear_model = XLinearModel.load(args.model_folder, is_predict_only=True)
-
-    if args.batch_size is not None:
-        Yts = []
-        for i in range(0, Xt.shape[0], args.batch_size):
-            Yte = xlinear_model.predict(
-                Xt[i : i + args.batch_size, :],
+    # Select Output
+    if args.select_output is not None:
```

Comment:
Line 129-152 are implementing the same logic as line 153-176. Can the code be reused? It seems the only difference is whether to provide `select_outputs_csr` as `None`.

Reply:
Function has been modified to reuse the same predict function.